fix(security): Resolve flaky SecureFileOps watch event tests

Part of #560 — Tier 4c: Test Failures

## Summary

3 tests in `tests/security/SecureFileOps.test.ts` fail due to **race conditions in filesystem watch event detection**. Additionally, the test file triggers an `EMFILE: too many open files` uncaught exception that affects the entire test suite run.

## Current State

3 failing tests:
```
FAIL tests/security/SecureFileOps.test.ts
  ✗ SecureFileOps > watch > should watch a directory for file changes
      AssertionError: expected false to be true
      → events.some((e) => e.path.includes('test.txt')).toBe(true)

  ✗ SecureFileOps > event types > should emit change event for modified files
      AssertionError: expected 0 to be greater than 0
      → modifyEvents.length.toBeGreaterThan(0)

  ✗ SecureFileOps > event types > should emit unlink event for deleted files
      AssertionError: expected 0 to be greater than 0
      → unlinkEvents.length.toBeGreaterThan(0)
```

Unhandled error during test run:
```
Error: EMFILE: too many open files, watch
  ❯ FSEvent.FSWatcher._handle.onchange node:internal/fs/watchers:214:21
```

## What

### 1. Fix the race condition in watch tests

The tests create files and immediately check for watch events. Filesystem watchers need time to initialize and events need time to propagate. Add proper wait mechanisms:

```typescript
// Instead of:
await writeFile(filePath, 'content');
expect(events.some(e => e.path.includes('test.txt'))).toBe(true);

// Use:
await writeFile(filePath, 'content');
await vi.waitFor(() => {
    expect(events.some(e => e.path.includes('test.txt'))).toBe(true);
}, { timeout: 2000 });
```

### 2. Fix the EMFILE leak

The `EMFILE: too many open files` error indicates file watchers are not being closed after each test. Ensure:
- Every `watch()` handle is closed in `afterEach`
- Use a test-scoped watcher limit
- Add cleanup in the `finally` block

### 3. Consider platform-specific behavior

macOS `FSEvents` and Linux `inotify` have different event timing. Tests should be resilient to platform differences:
- Use polling-based wait instead of fixed timeouts
- Allow for debounced events (some watchers batch events)

## Risks

| Risk | Severity | Mitigation |
|------|----------|------------|
| Fix masks real event delivery bugs | LOW | Keep event assertions — just add proper wait |
| EMFILE affects other tests in the suite | MEDIUM | Fix watcher cleanup — isolates the issue |
| Platform-specific test instability | LOW | Use `vi.waitFor()` with reasonable timeout |

## Acceptance Criteria

- [ ] All 3 `SecureFileOps` watch tests pass consistently (run 5x with no flakes)
- [ ] `EMFILE: too many open files` error no longer occurs during test run
- [ ] All file watcher handles are properly closed in test teardown
- [ ] `npm test` uncaught error count drops from 1 to 0

--- a/src/security/SecureFileOps.ts
+++ b/src/security/SecureFileOps.ts
@@ -750,7 +750,10 @@ export class SecureFileOps {
       }
     );
 
+    // Attach error handler immediately to prevent unhandled EMFILE errors
+    // from crashing the process when file descriptor limits are reached
     fsWatcher.on('error', (error) => {
+      watcherState.active = false;
       const event: FileWatchEvent = {
         type: 'error',
         path: relativePath,
@@ -791,7 +794,11 @@ export class SecureFileOps {
     }
 
     watcherState.active = false;
-    watcherState.watcher.close();
+    try {
+      watcherState.watcher.close();
+    } catch {
+      // Watcher may already be closed due to error (e.g., EMFILE)
+    }
 
     // Clear all pending debounce timers
     for (const timer of watcherState.debounceTimers.values()) {

--- a/tests/security/SecureFileOps.test.ts
+++ b/tests/security/SecureFileOps.test.ts
@@ -42,14 +42,17 @@ describe('SecureFileOps', () => {
       expect(handle.isActive()).toBe(true);
       expect(handle.watchPath).toBe('.');
 
+      // Wait for watcher to initialize before creating the file
+      await new Promise((resolve) => setTimeout(resolve, 50));
+
       // Create a file
       fs.writeFileSync(testFile, 'test content');
 
-      // Wait for debounce
-      await new Promise((resolve) => setTimeout(resolve, 100));
-
-      expect(events.length).toBeGreaterThan(0);
-      expect(events.some((e) => e.path.includes('test.txt'))).toBe(true);
+      // Poll until the event is received instead of fixed timeout
+      await vi.waitFor(() => {
+        expect(events.length).toBeGreaterThan(0);
+        expect(events.some((e) => e.path.includes('test.txt'))).toBe(true);
+      }, { timeout: 2000, interval: 50 });
 
       handle.close();
       expect(handle.isActive()).toBe(false);
@@ -305,18 +308,18 @@ describe('SecureFileOps', () => {
         events.push(event);
       }, { debounceMs: 10 });
 
-      // Wait a bit before modifying
-      await new Promise((resolve) => setTimeout(resolve, 50));
+      // Wait for watcher to initialize
+      await new Promise((resolve) => setTimeout(resolve, 100));
 
       fs.writeFileSync(testFile, 'modified');
 
-      await new Promise((resolve) => setTimeout(resolve, 150));
-
-      // Should have events for the modification (change or add depending on OS)
-      const modifyEvents = events.filter(
-        (e) => (e.type === 'change' || e.type === 'add') && e.path.includes('modify-test')
-      );
-      expect(modifyEvents.length).toBeGreaterThan(0);
+      // Poll until modification event is received
+      await vi.waitFor(() => {
+        const modifyEvents = events.filter(
+          (e) => (e.type === 'change' || e.type === 'add') && e.path.includes('modify-test')
+        );
+        expect(modifyEvents.length).toBeGreaterThan(0);
+      }, { timeout: 2000, interval: 50 });
 
       handle.close();
     });
@@ -331,17 +334,18 @@ describe('SecureFileOps', () => {
         events.push(event);
       }, { debounceMs: 10 });
 
-      // Wait a bit before deleting
-      await new Promise((resolve) => setTimeout(resolve, 50));
+      // Wait for watcher to initialize
+      await new Promise((resolve) => setTimeout(resolve, 100));
 
       fs.unlinkSync(testFile);
 
-      await new Promise((resolve) => setTimeout(resolve, 150));
-
-      const unlinkEvents = events.filter(
-        (e) => e.type === 'unlink' && e.path.includes('delete-test')
-      );
-      expect(unlinkEvents.length).toBeGreaterThan(0);
+      // Poll until unlink event is received
+      await vi.waitFor(() => {
+        const unlinkEvents = events.filter(
+          (e) => e.type === 'unlink' && e.path.includes('delete-test')
+        );
+        expect(unlinkEvents.length).toBeGreaterThan(0);
+      }, { timeout: 2000, interval: 50 });
 
       handle.close();
     });

